cpp/include/copc-lib/copc/file.hpp

```diff
@@ -13,23 +17,31 @@ const int COPC_OFFSET = 429;
 class CopcFile : public las::LasFile
 {
   public:
-    CopcFile(las::LasHeader header, las::CopcVlr copc, las::WktVlr wkt, las::EbVlr eb)
-        : LasFile(header, eb), copc(copc), wkt(wkt){};
-    CopcFile(las::LasHeader header, int span, std::string wkt, las::EbVlr eb) : LasFile(header, eb)
-    {
-        this->wkt.wkt = wkt;
-        this->copc.span = span;
-    };
+    CopcFile(const las::LasHeader &header, const CopcInfo &copc_info, const CopcExtents &copc_extents,
+             const std::string &wkt, const las::EbVlr &extra_bytes_vlr)
+        : LasFile(header, extra_bytes_vlr), copc_info_(copc_info), copc_extents_(copc_extents), wkt_(wkt){};
 
     // WKT string if defined, else empty
-    std::string GetWkt() const { return wkt.wkt; }
+    std::string GetWkt() const { return wkt_; }
+
+    // CopcInfo
+    CopcInfo GetCopcInfo() const { return copc_info_; }
 
-    // CopcData
-    las::CopcVlr GetCopc() const { return copc; }
+    // CopcExtents
+    CopcExtents GetCopcExtents() const { return copc_extents_; }
+
+    void SetCopcInfo(const CopcInfo &info) { copc_info_ = info; }
+    void SetCopcHier(uint64_t root_hier_offset, uint64_t root_hier_size)
```

Comment:
Why DOn't we just remove these two functions and return a shared pointer to CopcInfo?